### PR TITLE
Darkie changes

### PR DESCRIPTION
**Pull request by TC-DarkSlayer**
Since "4x bullet" bug was supposedly fixed, armor durability was scaled back down again, but its durability is still better than previous update.
Reduced knockback from shotguns by 30% as shotguns shoots a lot more bullets than other guns, causing people to fly in the air due to the sheer knockback.

### CODE REVIEW

**Comment by xcs011** [comment]
literally when you crackhead
unless it was fixed in actually the last 8 hours it didnt happen

**Comment by laikadoggirl** [comment]
also why do you have two separate githubs to make the pr with one and commit with the other wth

**Comment by TC-DarkSlayer** [comment]
> also why do you have two separate githubs to make the pr with one and commit with the other wth

Forgot account pass then remembered
Unless you're talking about why I have 3 branches

**Comment by laikadoggirl** [comment]
Also @TC-DarkSlayer the knockback change is no longer needed, it was due to the 4x damage glitch as well

Shotguns also shot 4x the number of pellets, and had 4x the knockback

File: TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/MilitaryHelmet/militaryhelmet_effect.as [hunk 1]
```diff
@@ -58,7 +58,7 @@ void onTick(CBlob@ this)
         headoffset += Vec2f(-head_offset.x, head_offset.y);
         headoffset += Vec2f(0, -1);
         milhelmet.SetOffset(headoffset);
-		milhelmet.SetFrameIndex(Maths::Floor(this.get_f32("militaryhelmet_health") / 10.01f));
+		milhelmet.SetFrameIndex(Maths::Floor(this.get_f32("militaryhelmet_health") / 7.51f));
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/MilitaryHelmet/militaryhelmet_effect.as]: indent

**Review by Vam-Jam — CHANGES_REQUESTED** [review]

**Comment by Vam-Jam** [comment]
Holding this back an update (will include in a hotfix). Waiting to see if bullet damage is finally fixed

File: TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as [hunk 2]
```diff
@@ -97,12 +97,17 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 		
 		if (headname != "" && this.exists(headname+"_health"))
 		{
-			f32 armorMaxHealth;
-			f32 ratio;
+			f32 armorMaxHealth=10.0f;
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as]: armorMaxHealth = 10.0f, space for readability

File: TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as [hunk 3]
```diff
@@ -141,12 +143,14 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 		}
 		if (torsoname != "" && this.exists(torsoname+"_health"))
 		{
-			f32 armorMaxHealth;
-			f32 ratio;
+			f32 armorMaxHealth=10.0f;
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as]: same here

File: TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as [hunk 4]
```diff
@@ -188,12 +191,11 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 
 		if (bootsname != "" && this.exists(bootsname+"_health"))
 		{
-			f32 armorMaxHealth;
-			f32 ratio;
-
+			f32 armorMaxHealth=10.0f;
+			f32 ratio=0.0f;
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as]: and here

**Review by Vam-Jam — CHANGES_REQUESTED** [review]

**Comment by Vam-Jam** [comment]
also fix merge conflicts if you can

**Comment by TC-DarkSlayer** [comment]
This PR was based on a PR that was reverted
Requires this revert to be reverted 
image

**Review by TC-DarkSlayer — COMMENTED** [review]
- Building Druglab gives 33% chance to give you chemistry blueprint
(may tweak percent chances later on, we'll see for now how this gacha mechanic goes)

**Comment by TC-DarkSlayer** [comment]
Current Summary:
	Code Stuff: indents, and making things use switch case to make it faster
	New Armor : Scubagear, minershelmet, pumpkin, combatboots (now protects minor damage)
	Ballistic Armor now has more durability
	Shotgun 30% less knockback
	Repairable Armor and Bushyhelmet(camo) upgrade
	Tinkertable now sorted
	Lighter costs a lantern instead of oil
	Grinder gives a little bit more wood
	Placing backwalls gives coins
	Food is now stacked into one blob, reducing lag
	Juggernaut can now use V to eat
	PlayerChickens can now throw grenades
	Fixed teamstorage bugs
	Fixed double count faction upgrading
	Upgrading/Building/Repairing faction bases now costs less as a result of above
	Armory gives teamStorage remote access
	Making Druglabs has chance of dropping blueprint: Chemistry
	Blueprints will no longer be consumed
	Royal guard can use flippers
	GrainPlants now drop 1-3 grain
	Grain Stacks to 10
	Badgers drop less meat
	Can craft coal from forges
	Food cans costs 2x more
	Reduced bullet resistance of dirt
	Engineer tools can be made via tinkertable
	Foof AoE effect caps at 5 foof
	Domino and crak speed effect reduced
	Dangerous Domino Recipe buffed
	Fumes duration buffed
	Crak no longer has damage reduction
	KudzuRework:
		KudzuCores can now get specific mutations by feeding the following:
		mithril ingot/ore/enriched, dirt, oil, meat, steak, goldore, badger, and sulphur
		Mithril ingots are used to hasten the mutation rates
		KudzuCores no longer has damage reduction from bullets/bombs
		KudzuCores can now be damaged by radiation
		Kudzu now can mutate radiation resistance
		KuzuCores costs 5 mithril ingots
		Kudzu no longer has damage reduction from bullets/bombs
		Kudzu can now be damaged by radiation

**Comment by Wunarg** [comment]
Really not a fan of the kudzu damage resistances changes, since defeating kudzu with bombs and bullets removes one of the intersting aspects of kudzu of needing to use other weaponary, instead of the weaponary used everywhere else

**Comment by TC-DarkSlayer** [comment]
> Really not a fan of the kudzu damage resistances changes, since defeating kudzu with bombs and bullets removes one of the intersting aspects of kudzu of needing to use other weaponary, instead of the weaponary used everywhere else

Was pain in a neck to destroy kudzu cores, took me around 8 dynamite per kudzucore(kudzu tiles kept regening) vs mason's 15 kudzu cores

**Comment by Wunarg** [comment]
> 
> 
> > Really not a fan of the kudzu damage resistances changes, since defeating kudzu with bombs and bullets removes one of the intersting aspects of kudzu of needing to use other weaponary, instead of the weaponary used everywhere else
> 
> Was pain in a neck to destroy kudzu cores, took me around 8 dynamite per kudzucore(kudzu tiles kept regening) vs mason's 15 kudzu cores

thats what i mean though, your not supposed to use explosives against kudzu

**Comment by laikadoggirl** [comment]
I don't think shotgun needs a further knockback nerf, as that was majorly fixed when the 4x bullets bug was.
It would also almost make more sense to just make the chemlab have a chance to drop the blueprint when created, like you got experience from building it, and made the blueprint

**Comment by laikadoggirl** [comment]
Reducing dirt bullet resistance is eh too, rather I think it should be like neutrals with concrete, where they can damage it, but never fully break it

**Comment by charliehstreet** [comment]
neuts can completely break concrete, you're thinking of reinforced steel concrete. Also autochemlab should drop its blueprint everytime its destroyed, along with the missile assembler and the sun tower

**Comment by laikadoggirl** [comment]
Oh, I literally meant that the normal chem lab should have a chance to drop it on construction, albeit a small chance, and yeah I was thinking about reinforced

**Comment by laikadoggirl** [comment]
That or ill finally add a sandbag block to absorb bullets

**Comment by laikadoggirl** [comment]
> 
> 
> Really not a fan of the kudzu damage resistances changes, since defeating kudzu with bombs and bullets removes one of the intersting aspects of kudzu of needing to use other weaponary, instead of the weaponary used everywhere else

I mean yeah but why wouldnt a plant be just as vulnerable to explosives

File: TFlippy_TerritoryControl_Misc_Dev/Entities/Misc/Kudzu/KudzuCore.as [hunk 5]
```diff
@@ -71,23 +95,43 @@ void onCommand(CBlob@ this, u8 cmd, CBitStream @params) //Mutate command
...
+				{
+					case -661490310:	matReq = 10;	break;	// mithrilingot
+					case -989285105:	matReq = 10;	break;	// mithrilenriched
+					case -1288560969:	matReq = 200; 	break;	// mithril
+					case 1074492747:	matReq = 500; 	break;	// dirt
+					case -1326479778:	matReq = 50; 	break;	// oil
+					case -123101143: 	matReq = 100;	break;	// meat
+					case -1370030172:	matReq = 250;	break;	// gold ore
+					case -617913447:	matReq = 250;	break;	// sulphur
+					case 389592510:		matReq = 1;		break;	// badger
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Misc_Dev/Entities/Misc/Kudzu/KudzuCore.as]: indent

File: TFlippy_TerritoryControl_Misc_Dev/Entities/Misc/Kudzu/KudzuCore.as [hunk 6]
```diff
@@ -54,12 +59,31 @@ const u32 RECHARGETIME = 18000; // 30 = 1 second, 1800 = 1 minute, this is 10 mi
...
+			case 1074492747:	// dirt
+			case -1326479778:	// oil
+			case -123101143:	// meat
+			case -1370030172:	// gold ore
+			case -617913447:	// sulphur
+			case 389592510:		// badger
+			case 336243301:		// steak
+				canMutate = true;
+				break;
+		}
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Misc_Dev/Entities/Misc/Kudzu/KudzuCore.as:81]: space

**Review by Vam-Jam — CHANGES_REQUESTED** [review]

**Comment by Vam-Jam** [comment]
This PR has gotten too big, I'll sit down and review it when I get the time.

**Comment by TC-DarkSlayer** [comment]
Splitting this PR

**Pull request closed without merging**